test/integration/restart_task_test.rb

```diff
@@ -0,0 +1,108 @@
+
+class RestartTaskTest < KubernetesDeploy::IntegrationTest
+  def test_restart
+    deploy_fixtures("basic", subset: ["configmap-data.yml", "web.yml.erb"])
+
+    deployment = fetch_deployment("web")
+    refute fetch_restarted_at(deployment), "no RESTARTED_AT env on fresh deployment"
+
+    restart = KubernetesDeploy::RestartTask.new(
+      context: KubeclientHelper::MINIKUBE_CONTEXT,
+      namespace: @namespace,
+    )
+    restart.perform(["web"])
+
+    assert_logs_match(/Triggered "web" restart/)
+    assert_logs_match(/Successfully restarted "web"/)
+
+    deployment = fetch_deployment("web")
+    assert fetch_restarted_at(deployment), "RESTARTED_AT is present after the restart"
+  end
+
+  def test_restart_twice
+    deploy_fixtures("basic", subset: ["configmap-data.yml", "web.yml.erb"])
+
+    deployment = fetch_deployment("web")
+    refute fetch_restarted_at(deployment), "no RESTARTED_AT env on fresh deployment"
+
+    restart = KubernetesDeploy::RestartTask.new(
+      context: KubeclientHelper::MINIKUBE_CONTEXT,
+      namespace: @namespace,
+    )
+    restart.perform(["web", "web"])
+
+    assert_logs_match_once(/Triggered "web" restart/)
+    assert_logs_match_once(/Successfully restarted "web"/)
+
+    deployment = fetch_deployment("web")
+    assert fetch_restarted_at(deployment), "RESTARTED_AT is present after the restart"
+  end
+
+  def test_restart_not_existing_deployment
+    restart = KubernetesDeploy::RestartTask.new(
+      context: KubeclientHelper::MINIKUBE_CONTEXT,
+      namespace: @namespace,
+    )
+    assert_raises(KubernetesDeploy::RestartTask::DeploymentNotFoundError) do
+      restart.perform(["web"])
+    end
+  end
+
+  def test_restart_one_not_existing_deployment
+    restart = KubernetesDeploy::RestartTask.new(
+      context: KubeclientHelper::MINIKUBE_CONTEXT,
+      namespace: @namespace,
+    )
+    assert_raises(KubernetesDeploy::RestartTask::DeploymentNotFoundError) do
+      restart.perform(["walrus", "web"])
+    end
+
+    deploy_fixtures("basic", subset: ["configmap-data.yml", "web.yml.erb"])
+    deployment = fetch_deployment("web")
+    refute fetch_restarted_at(deployment), "no RESTARTED_AT env after failed restart task"
+  end
+
+  def test_restart_none
+    restart = KubernetesDeploy::RestartTask.new(
+      context: KubeclientHelper::MINIKUBE_CONTEXT,
+      namespace: @namespace,
+    )
+    assert_raises(ArgumentError) do
+      restart.perform([])
+    end
+  end
+
+  def test_restart_not_existing_context
+    assert_raises(KubernetesDeploy::KubeclientBuilder::ContextMissignError) do
+      KubernetesDeploy::RestartTask.new(
+        context: "walrus",
+        namespace: @namespace,
+      )
+    end
+  end
+
+  def test_restart_not_existing_namespace
+    restart = KubernetesDeploy::RestartTask.new(
+      context: KubeclientHelper::MINIKUBE_CONTEXT,
+      namespace: "walrus"
+    )
+    assert_raises(KubernetesDeploy::RestartTask::DeploymentNotFoundError) do
```

Comment:
Hm, I think the distinction between the namespace not existing and the deployment not existing is very useful to the user. Can we validate the namespace before attempting to access the deployment so we can make that distinction?